tests/fe/point_evaluation_04.output

```diff
@@ -1,310 +1,310 @@
+DEAL::-0.02139651817 0.7271443718: -0.02139651817 1.727144372 error value 2.227212005e-16 error grad 6.578664217e-16 error scalar value -2.220446049e-16 error scalar grad 5.144514810e-16
+DEAL::-0.06159439879 0.6951573525: -0.06159439879 1.695157352 error value 2.221529987e-16 error grad 3.385178120e-15 error scalar value 2.220446049e-16 error scalar grad 6.604949552e-16
+DEAL::-0.09833207180 0.6609086767: -0.09833207180 1.660908677 error value 2.220446049e-16 error grad 1.488978900e-15 error scalar value 0.000000000 error scalar grad 4.484029338e-16
+DEAL::-0.1316095372 0.6247255774: -0.1316095372 1.624725577 error value 2.237726046e-16 error grad 1.109285710e-15 error scalar value 2.220446049e-16 error scalar grad 1.574435834e-15
 DEAL::
-DEAL::18.89104831 18.67281236 0.000000000 -0.6801424008 2.148833945 0.000000000 -0.1829900258 -1.747627068 0.000000000 -0.1385764851 0.3355885892 0.000000000 0.9524375032 -1.671256175 -29.90839999 7.629284927 0.000000000 2.346228990 -2.665568356 -5.053709590 1.174822557 0.000000000 -5.143470224 4.086106873 24.87161852 -9.859895741 0.000000000 -1.636339156 3.204486273 4.330598384 -5.502429773 0.000000000 -2.301228679 -16.82258884 16.14732884 -12.51113875 19.81001935 30.56188453 -25.42295303 -8.883950370 0.000000000 
-DEAL::0.000000000 0.000000000 22.03790992 0.000000000 0.000000000 0.7973538527 0.000000000 0.000000000 -0.4976854603 0.000000000 0.000000000 -0.9325339590 0.000000000 0.000000000 0.000000000 0.000000000 8.927524173 0.000000000 0.000000000 0.000000000 0.000000000 -9.189558667 0.000000000 0.000000000 0.000000000 0.000000000 -3.881641732 0.000000000 0.000000000 0.000000000 0.000000000 4.406210129 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 6.206759441 
+DEAL::18.89104831 18.67281236 2.500000000 -0.6801424008 2.148833945 1.500000000 -0.1829900258 -1.747627068 2.250000000 -0.1385764851 0.3355885892 1.750000000 0.9524375032 -1.671256175 -29.90839999 7.629284927 2.375000000 2.346228990 -2.665568356 -5.053709590 1.174822557 1.625000000 -5.143470224 4.086106873 24.87161852 -9.859895741 2.000000000 -1.636339156 3.204486273 4.330598384 -5.502429773 2.000000000 -2.301228679 -16.82258884 16.14732884 -12.51113875 19.81001935 30.56188453 -25.42295303 -8.883950370 2.000000000 
+DEAL::18.89104831 18.67281236 22.03790992 -0.6801424008 2.148833945 0.7973538527 -0.1829900258 -1.747627068 -0.4976854603 -0.1385764851 0.3355885892 -0.9325339590 0.9524375032 -1.671256175 -29.90839999 7.629284927 8.927524173 2.346228990 -2.665568356 -5.053709590 1.174822557 -9.189558667 -5.143470224 4.086106873 24.87161852 -9.859895741 -3.881641732 -1.636339156 3.204486273 4.330598384 -5.502429773 4.406210129 -2.301228679 -16.82258884 16.14732884 -12.51113875 19.81001935 30.56188453 -25.42295303 -8.883950370 6.206759441 
```

Comment:
This is one thing we should discuss: The optimized path only writes into those DoFs where the selected component is non-zero, leaving the rest unchanged. The old implementation did zero the other parts. What do we want to have? I would find it more natural to only touch those that belong to the selected part, but it would one more round of checks with `FEValues`. Or I just add a zeroing operation to the tensor product path.

Reply:
I now pushed a new version where the behavior is clearly documented (we clear `solution_values` in that function, consistent with #12235). The new code is hence consistent with all previous tests, showing that the new functionality is correct.